src/lock/mod.rs

```diff
@@ -77,10 +77,20 @@ pub fn config(ctx: &Context, config: Config) -> Result<LockedConfig> {
 
     // Create a map of unique `Source` to `Vec<Plugin>`
     let mut map = IndexMap::new();
+    let profile = std::env::var("SHELDON_PROFILE");
```

Comment:
Instead of looking up the value here I think it would be better to add it as a field to the `Context` struct. And lookup the environment variable when the `Context` is constructed. Like this:

```rust
pub struct Context {
    ...
    pub clone_dir: PathBuf,
    pub download_dir: PathBuf,
    pub profile: String,
    ...
}
```

You should also add it to the `is_context_equal` function so that if the profile changes then sheldon will regenerate the lock file.

We should also add a `--profile` option to the Sheldon CLI so that you can force run it with a particular profile, but I'm happy if that's done in a different PR.